src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -114,84 +101,86 @@ public bool ContentSet
         /// <summary>
         /// Generates the <see cref="TagHelperOutput"/>'s start tag.
         /// </summary>
-        /// <returns><c>string.Empty</c> if <see cref="TagName"/> is <c>null</c> or whitespace. Otherwise, the
+        /// <returns><c>Null</c> if <see cref="TagName"/> is <c>null</c> or whitespace. Otherwise, the
         /// <see cref="string"/> representation of the <see cref="TagHelperOutput"/>'s start tag.</returns>
         public string GenerateStartTag()
```

Comment:
- as I mentioned elsewhere, "really long text" is a worst case. we aim to avoid all useless `string` concatenations we can.
- this is about returning the same type as e.g. `GenerateContent()`.  that's an `ITextWriterCopyable` (aka `TagHelperContent`) instance, not a `TextWriter`.